fl4health/parameter_exchange/sparse_coo_parameter_exchanger.py

```diff
@@ -12,117 +12,114 @@
-        Args:
-            model (nn.Module): Current model.
-            initial_model (nn.Module): Initial model.
-
-        Returns:
-            Dict[str, Tensor]: Dictionary that maps the name of each tensor to its corresponding mask.
-        """
-        all_parameter_scores = self.generate_parameter_scores(model, initial_model)
-        all_scores = torch.cat([val.flatten() for _, val in all_parameter_scores.items()])
-        # Sorting all scores and determining the threshold.
-        sorted_scores, _ = torch.sort(all_scores, descending=True)
-        top_scores = sorted_scores[: math.ceil(len(sorted_scores) * self.sparsity_level)]
-        score_threshold = top_scores[-1].item()
-
-        names_to_masks = {}
-        for name, param_scores in all_parameter_scores.items():
-            # Use score_threshold to produce masks.
-            threshold_result = torch.where(param_scores >= score_threshold, input=param_scores, other=0)
-            mask = torch.where(threshold_result == 0, input=threshold_result, other=1)
-            # Tensor whose mask values are all zero will not be exchanged, so we
-            # do not return them.
-            if not (mask == 0).all():
-                names_to_masks[name] = mask
-        return names_to_masks
-
-    def select_parameters(
-        self, model: nn.Module, initial_model: Optional[nn.Module] = None
-    ) -> Tuple[NDArrays, Tuple[NDArrays, NDArrays, List[str]]]:
-        """
-        Generate masks for model's parameters and select those with mask values 1
-        for exchange. Also extract all the information required to represent
+        Finally, the method extracts all the information required to represent
         the selected parameters in the sparse COO tensor format. More specifically,
         the information consists of the indices of the parameters within the tensor
         to which they belong, the shape of that tensor, and also the name of it.
 
         Args:
             model (nn.Module): Current model.
-            initial_model (Optional[nn.Module], optional): Initial model. Defaults to None.
+            initial_model (nn.Module): Initial model.
 
         Returns:
             Tuple[NDArrays, Tuple[NDArrays, NDArrays, List[str]]]: the selected parameters
             and other information, as detailed above.
         """
-        tensor_names_to_masks = self.create_masks(model, initial_model)
-        model_states = model.state_dict()
+        all_parameter_scores = self.generate_parameter_scores(model, initial_model)
+        all_scores = torch.cat([val.flatten() for _, val in all_parameter_scores.items()])
+        # Sorting all scores and determining the threshold.
+        sorted_scores, _ = torch.sort(all_scores, descending=True)
+        n_top_scores = math.ceil(len(sorted_scores) * self.sparsity_level)
+        # Sanity check.
+        assert n_top_scores >= 1
+        top_scores = sorted_scores[:n_top_scores]
+        score_threshold = top_scores[-1].item()
```

Comment:
Perhaps, I'm wrong, but could we just do 
```
score_threshold = sorted_scores[n_top_scores-1]
```
rather than slicing sorted_scores and taking the last element?

Reply:
Yes absolutely